org/mozilla/jss/util/NativeProxy.java

```diff
@@ -88,19 +99,54 @@ public boolean equals(Object obj) {
      *      }
      * }
      *
-     * @deprecated finalize() in Object has been deprecated
+     * @deprecated finalize() in Object has been deprecated. Use close(...)
+     * from the AutoCloseable interface instead.
      */
     @Deprecated
     protected void finalize() throws Throwable {
-        unregister(registryIndex);
-        releaseNativeResources();
+        close();
+    }
+
+    /**
+     * Close this NativeProxy by releasing its native resources if they
+     * haven't otherwise been freed.
+     *
+     * See comment in finalize.
+     */
+    public final void close() throws Exception {
+        try {
+            if (unregister(this)) {
+                releaseNativeResources();
+            }
+        } catch (Exception e) {
+            throw e;
```

Comment:
I think we can drop the catch clause since it's simply rethrowing the exception.